command/pipe.go

```diff
@@ -0,0 +1,328 @@
+	"github.com/peak/s5cmd/v2/log"
+	"github.com/peak/s5cmd/v2/log/stat"
+	"github.com/peak/s5cmd/v2/storage"
+	"github.com/peak/s5cmd/v2/storage/url"
+)
+
+var pipeHelpTemplate = `Name:
+	{{.HelpName}} - {{.Usage}}
+
+Usage:
+	{{.HelpName}} [options] destination
+
+Options:
+	{{range .VisibleFlags}}{{.}}
+	{{end}}
+Examples:
+	01. Stream stdin output to an object
+		 > echo "content" | gzip | s5cmd {{.HelpName}} s3://bucket/prefix/object.gz
+`
+
+func NewPipeCommandFlags() []cli.Flag {
+	pipeFlags := []cli.Flag{
+		&cli.StringFlag{
+			Name:  "storage-class",
+			Usage: "set storage class for target ('STANDARD','REDUCED_REDUNDANCY','GLACIER','STANDARD_IA','ONEZONE_IA','INTELLIGENT_TIERING','DEEP_ARCHIVE')",
+		},
+		&cli.IntFlag{
+			Name:    "concurrency",
+			Aliases: []string{"c"},
+			Value:   defaultCopyConcurrency,
+			Usage:   "number of concurrent parts transferred between host and remote server",
+		},
+		&cli.IntFlag{
+			Name:    "part-size",
+			Aliases: []string{"p"},
+			Value:   defaultPartSize,
+			Usage:   "size of each part transferred between host and remote server, in MiB",
+		},
+		&cli.StringFlag{
+			Name:  "sse",
+			Usage: "perform server side encryption of the data at its destination, e.g. aws:kms",
+		},
+		&cli.StringFlag{
+			Name:  "sse-kms-key-id",
+			Usage: "customer master key (CMK) id for SSE-KMS encryption; leave it out if server-side generated key is desired",
+		},
+		&cli.StringFlag{
+			Name:  "acl",
+			Usage: "set acl for target: defines granted accesses and their types on different accounts/groups, e.g. pipe --acl 'public-read'",
+		},
+		&cli.StringFlag{
+			Name:  "cache-control",
+			Usage: "set cache control for target: defines cache control header for object, e.g. pipe --cache-control 'public, max-age=345600'",
+		},
+		&cli.StringFlag{
+			Name:  "expires",
+			Usage: "set expires for target (uses RFC3339 format): defines expires header for object, e.g. pipe  --expires '2024-10-01T20:30:00Z'",
+		},
+		&cli.BoolFlag{
+			Name:  "raw",
+			Usage: "disable the wildcard operations, useful with filenames that contains glob characters",
+		},
+		&cli.StringFlag{
+			Name:  "content-type",
+			Usage: "set content type for target: defines content type header for object, e.g. --content-type text/plain",
+		},
+		&cli.StringFlag{
+			Name:  "content-encoding",
+			Usage: "set content encoding for target: defines content encoding header for object, e.g. --content-encoding gzip",
+		},
+		&cli.StringFlag{
+			Name:  "content-disposition",
+			Usage: "set content disposition for target: defines content disposition header for object, e.g. --content-disposition 'attachment; filename=\"filename.jpg\"'",
+		},
+		&cli.IntFlag{
+			Name:        "no-such-upload-retry-count",
+			Usage:       "number of times that a request will be retried on NoSuchUpload error; you should not use this unless you really know what you're doing",
+			DefaultText: "0",
+			Hidden:      true,
+		},
```

Comment:
We should also remove this flag. 

As standart input is not seek-able, on such error it might be fully or partially consumed. To handle it we need to wait for standart input to be closed but doing makes the command behaves exactly like `cp` -with all contents in memory- that invalidates the reason we have this command in the first place.

> PS: We've seem to overlooked the case where reader would already be consumed when this option first introduced, it looks buggy anyways. I'll try to test and open a PR if necessary.

Reply:
Removed. https://github.com/peak/s5cmd/pull/587/commits/bb450a131c93afa69403f9a8196ec989774dee61